jupyterhub/orm.py

```diff
@@ -716,18 +716,8 @@ def new(
             orm_token.expires_at = cls.now() + timedelta(seconds=expires_in)
 
         db.add(orm_token)
-        token_role = Role.find(db, 'token')
-        if not token_role:
-            # FIXME: remove this.
-            # Creating a token before the db has roles defined should raise an error.
-            # PR #3460 should let us fix it by ensuring default roles are defined
-
-            warnings.warn(
-                "Token created before default roles!", RuntimeWarning, stacklevel=2
-            )
-            default_roles = get_default_roles()
-            for role in default_roles:
-                create_role(db, role)
+        if not Role.find(db, 'token'):
+            raise AttributeError("Default token role has not been created")
```

Comment:
Let's call this RuntimeError since it means the state of the process is generally wrong. AttributeError is only for failure to get a specific attribute of an object.

Reply:
:+1: